backend/src/services/model.ts

```diff
@@ -335,6 +342,39 @@ export async function updateModel(user: UserInterface, modelId: string, modelDif
   return model
 }
 
+async function checkCollaboratorAuthorisation(collaborators: CollaboratorEntry[]) {
+  const duplicateEntities = collaborators.reduce<string[]>(
+    (duplicates, currentCollaborator, currentCollaboratorIndex) => {
+      if (
+        collaborators.find(
+          (collaborator, index) =>
+            index !== currentCollaboratorIndex && collaborator.entity === currentCollaborator.entity,
+        ) &&
+        !duplicates.includes(currentCollaborator.entity)
+      ) {
+        duplicates.push(currentCollaborator.entity)
+      }
+      return duplicates
+    },
+    [],
+  )
+  if (duplicateEntities.length > 0) {
+    throw BadReq(`The following duplicate collaborators have been found: ${duplicateEntities.join(', ')}`)
+  }
+  await Promise.all(
+    collaborators.map(async (collaborator) => {
+      if (collaborator.entity === '') {
+        throw BadReq('Collaborator name must be a valid string')
+      }
+      try {
+        await authentication.getUserInformation(collaborator.entity)
```

Comment:
Ideally it would be good to only do this check on just the collaborators that have been added to save having to make repeated external calls for entities that have already been validated. However, this would mean having to do a DB call to get the existing collaborators array so you could diff it with this one